mslice/presenters/powder_projection_presenter.py

```diff
@@ -57,10 +57,11 @@ def _calculate_powder_projection(self):
     def calc_projection(self, workspace, axis1, axis2, units):
         return self._projection_calculator.calculate_projection(workspace, axis1, axis2, units)
 
-    def after_projection(self, projection_workspaces):
-        self._get_main_presenter().update_displayed_workspaces()
-        self._get_main_presenter().change_ws_tab(1)
-        self._get_main_presenter().set_selected_workspaces(projection_workspaces)
+    def after_projection(self, projection_workspaces, is_gui=True):
+        if is_gui:
```

Comment:
Again here. If the main_presenter was allowed to be `None` we could just test for that here and do nothing if  `self._get_main_presenter()` is `None` meaning we wouldn't need the `is_gui` flag. We could then move the `after_projection` call to `calc_projection` and remove it from `_cli_commands` so there is a consistent behavioue between the GUI/CLI.